React/Base/RCTImageSource.m

```diff
@@ -60,7 +60,8 @@ + (RCTImageSource *)RCTImageSource:(id)json
   CGFloat scale = 1.0;
   BOOL packagerAsset = NO;
   if ([json isKindOfClass:[NSDictionary class]]) {
-    if (!(imageURL = [self NSURL:json[@"uri"]])) {
+    id uri = json[@"uri"];
+    if (!([uri isKindOfClass:[NSString class]]) || !(imageURL = [self NSURL:uri])) {
```

Comment:
Can you change this to

    if (!(imageURL = [self NSURL:RCTNilIfNull(json[@"uri"]]))) {